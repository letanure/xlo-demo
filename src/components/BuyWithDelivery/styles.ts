import styled, { css } from 'styled-components'

import { BuyWithDeliveryProps } from '.'

export const Wrapper = styled.div<BuyWithDeliveryProps>`
  ${({ prop1 }) => css`
    content: '${prop1}';
  `}
`
