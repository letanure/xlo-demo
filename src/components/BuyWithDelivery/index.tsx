import * as S from './styles'
import PriceDetails, { PriceDetailsProps } from 'components/ui/PriceDetails'

export type BuyWithDeliveryProps = {
  title?: string | number
}

const BuyWithDelivery = ({ title = 'default' }: BuyWithDeliveryProps) => {
  const prices: PriceDetailsProps['prices'] = [
    {
      label: 'Item',
      price: {
        amount: 259900,
        currency: 'PLN'
      }
    },
    {
      label: 'DELIVERY FROM',
      price: {
        amount: 400,
        currency: 'PLN'
      },
      discount: {
        amount: 899,
        currency: 'PLN'
      },
      description:
        'This discount applies for deliveries with InPost. In the 30 days before the start of this discount, the lowest price was 8,99 z≈Ç. Check terms and conditions'
    },
    {
      label: 'SERVICE FEE',
      icon: 'info',
      price: {
        amount: 250,
        currency: 'PLN'
      }
    },
    {
      label: 'SAFE DEAL',
      tag: {
        text: 'FREE'
      },
      description:
        'You have 24 hours to confirm that the item you ordered was in the package. Was the package empty or contained the wrong item? Submit a ticket to get your money back. How the Protection Package works'
    }
  ]
  return (
    <S.Wrapper>
      <S.Title>{title}</S.Title>
      <PriceDetails prices={prices} />
    </S.Wrapper>
  )
}

export default BuyWithDelivery
