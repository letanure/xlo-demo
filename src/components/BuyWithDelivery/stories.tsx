import { ComponentStory, ComponentMeta } from '@storybook/react'

import BuyWithDelivery from '.'

export default {
  title: 'Layout/BuyWithDelivery',
  component: BuyWithDelivery,
  argTypes: {
    prop1: { control: 'text' }
  }
} as ComponentMeta<typeof BuyWithDelivery>

const Template: ComponentStory<typeof BuyWithDelivery> = (args) => (
  <BuyWithDelivery {...args} />
)

export const Basic = Template.bind({})
Basic.args = {
  prop1: 'something'
}
