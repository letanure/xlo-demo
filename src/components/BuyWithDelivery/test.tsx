import { render, screen } from '@testing-library/react'

import BuyWithDelivery from '.'

describe('<BuyWithDelivery />', () => {
  it('should render something', () => {
    const { container } = render(<BuyWithDelivery />)

    expect(
      screen.getByRole('heading', { name: /BuyWithDelivery/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
