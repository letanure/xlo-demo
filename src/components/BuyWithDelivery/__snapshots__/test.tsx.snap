// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BuyWithDelivery /> should render something 1`] = `
.c0 {
  content: 'default';
}

<div
  class="c0"
>
  <h1>
    BuyWithDelivery with 
    default
     2
  </h1>
</div>
`
